### DISCUSSION TITLE
Multi-tenancy disable automatic tenant scoping per resource

### DISCUSSION THREAD

**Discussion by impeto** [filamentphp/filament#8059] [Help] [answered]
This may be unique to my project, but is there a way to disable tenant scoping on some of the resources? My app delivers police incident reports to law firms. The incident reports get dumped into the database daily and each law firm filters, sorts and downloads the reports according to their individual needs, therefore they can't be scoped to a law firm. Basically they look at what's in the database and pick what they need.

Thanks

**Comment by danharrin** [comment]
Have a look in the base Resource class, theres probably a method you can override to prevent the scoping

**Reply by kurucu** [reply, marked as answer]
Try this on a resource page:

```php
public static function scopeEloquentQueryToTenant(Builder $query, ?Model $tenant): Builder
{
    return $query;
}
```

**Reply by impeto** [reply]
Yes. Thank you guys. `return $query` (without the parens)

**Reply by HermanCeaser** [reply]
How about if i want to create a record that is not tenant-aware. Forexample I want to create a permission that is global for all tenants. ` It shouldn't have tenant_id` on it. Which Method Should i Override

**Reply by impeto** [reply]
Off the top of my head I would look at `mutateFormDataBeforeCreate` method in the `CreatePermission` class, assuming that's how you are creating the resource.

**Reply by kurucu** [reply]
I did this by using a separate non-Tenant panel. All much easier to go this way in the end. And makes sense really, if your permissions are outside of tenants, then perhaps they are a part of an Admin panel, or a User panel, or similar?

**Reply by HermanCeaser** [reply]
The way i chose to do it is the laravel way. From my understanding Multitenancy in Filament uses Global scopes under the hood for all the model events.

So i overrode the `handleRecordCreation` in the `createPermission` class and muted all the events when creating a Permission

`$permission = static::getModel()::withoutEvents(fn() => static::getModel()::create($data));` that way it doesn't apply the tenant_id global scope on save.

**Reply by danharrin** [reply]
That would be an observer, not a global scope. And no, we dont use observers out of the box.

**Reply by HermanCeaser** [reply]
Thank you for the articulation @danharrin. What would be the recommended filament way of achieving this!!

**Reply by impeto** [reply]
@HermanCeaser Look at `Filament\Resources\Pages\CreateRecord::handleRecordCreation`:

```php
protected function handleRecordCreation(array $data): Model
    {
        $record = new ($this->getModel())($data);

        if ($tenant = Filament::getTenant()) {
            return $this->associateRecordWithTenant($record, $tenant);
        }

        $record->save();

        return $record;
    }

```

You can override it in your `CreatePermission` class to skip the `if` statement. Filament doesn't use query scoping here, it uses the relationships defined in the RelationshipManages. Since you are overriding a method anyway, it doesn't make sense to mute the model's events. You may want to use them in the future. The scoping will be applied when you view the list of permissions. I'm guessing you will have to use the technique in the answer to my initial question to get all the permissions.
